src/aiida/calculations/stash.py

```diff
@@ -0,0 +1,77 @@
+###########################################################################
+# Copyright (c), The AiiDA team. All rights reserved.                     #
+# This file is part of the AiiDA code.                                    #
+#                                                                         #
+# The code is hosted on GitHub at https://github.com/aiidateam/aiida-core #
+# For further information on the license, see the LICENSE.txt file        #
+# For further information please visit http://www.aiida.net               #
+###########################################################################
+""""""
+
+from aiida import orm
+from aiida.common.datastructures import CalcInfo
+from aiida.engine import CalcJob
+
+
+class StashCalculation(CalcJob):
+    """
+    Utility to stash files from a remote folder.
```

Comment:
not only remote since you also support SinglefileData (and hopefully FolderData)